services/user.service.js

```diff
@@ -140,6 +141,108 @@ const deleteUserByUsername = async (usernameParamData) => {
   }
 };
 
+
+const userSignUp = async (body) => {
+  const id = userUtils.generateUUID();
+  const username = body.username;
+  const email = body.email;
+  const rawPassword = body.password;
+
+  if (!username) {
+    return { status: 400, message: "username Field is Empty" };
+  }
+
+  if (!email) {
+    return { status: 400, message: "email Field is Empty" };
+  }
+
+  if (!rawPassword) {
+    return { status: 400, message: "password Field is Empty" };
+  }
+
+  if (!userUtils.isAlphaNumeric(username)) {
+    return {
+      status: 401,
+      message:
+        "New User's username is null or contains space or special character",
+    };
+  }
+
+  if (!userUtils.validateEmail(email)) {
+    return { status: 401, message: "New User's email is not valid" };
+  }
+
+  if (!userUtils.checkPassword(rawPassword)) {
+    return { status: 401, message: "password is less than 6 digit" };
+  }
+
+  const password = await userUtils.generateHashPassword(body.password);
+  const createdAt = userUtils.formatUnixTimestamp(Date.now());
+  const updatedAt = userUtils.formatUnixTimestamp(Date.now());
+
+  /* const newUserData = [id, username, email, password, createdAt, updatedAt];*/
+  const newUser = await userRepository.userSignUp(
+    id,
+    username,
+    email,
+    password,
+    createdAt,
+    updatedAt
+  );
+
+  if (newUser) {
+    return {
+      status: 201,
+      message: `User: ${username} is successfully created`,
+    };
+  }
+};
+
+const userLogIn = async (body) => {
+  const username = body.username;
+  const rawPassword = body.password;
+
+  if (!username) {
+    return { status: 400, message: "username Field is Empty" };
+  }
+
+  if (!rawPassword) {
+    return { status: 400, message: "password Field is Empty" };
+  }
+
+  if (!userUtils.isAlphaNumeric(username)) {
+    return {
+      status: 401,
+      message:
+        "New User's username is null or contains space or special character",
+    };
+  }
```

Comment:
If you already ensured no invalid username will be signed up, then you can skip this part.

Reply:
I changed this message to "New User's username contains space or special character". Yes, invalid user checking is already done before this code. I did my mistake in error message.